src/main/java/com/greenfox/tribesoflagopus/backend/model/entity/Kingdom.java

```diff
@@ -23,34 +24,34 @@
 @NoArgsConstructor
 public class Kingdom {
 
+  @OneToMany(mappedBy = "kingdom", cascade = CascadeType.ALL, orphanRemoval = true)
+  private final List<Troop> troops = new ArrayList<>();
+
+  @OneToMany(mappedBy = "kingdom", cascade = CascadeType.ALL, orphanRemoval = true)
+  private final List<Building> buildings = new ArrayList<>();
+
+  @OneToMany(mappedBy = "kingdom", cascade = CascadeType.ALL, orphanRemoval = true)
+  private final List<Resource> resources = new ArrayList<>();
+
   @Id
   @GeneratedValue(strategy = GenerationType.AUTO)
   private long id;
 
-  @Builder
-  public Kingdom(String name, Player player,
-          List<Troop> troops,
-          List<Building> buildings,
-          Location location,
-          List<Resource> resources) {
-    this.name = name;
-    this.player = player;
-    this.troops = troops;
-    this.buildings = buildings;
-    this.location = location;
-    this.resources = resources;
-  }
-
   private String name;
 
   @OneToOne
   @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
   @JsonIdentityReference(alwaysAsId = true)
-  private Player player;
+  @JsonProperty(value = "user_id")
```

Comment:
We shouldn't do this many data transformation in our Entity model. Instead the Entity model should simply represent what we store in the database, and we should create a DTO object with the JSON transformations.